test/IDE/complete_ambiguous.swift

```diff
@@ -473,3 +473,88 @@ func testBestSolutionGeneric() {
 // BEST_SOLUTION_FILTER_GEN-DAG: Keyword[self]/CurrNominal:     self[#Test1#]; name=self
 // BEST_SOLUTION_FILTER_GEN: End completions
 
+func testAmbiguousArgs() {
+  struct A {
+    func someFunc(a: Int, b: Int) -> A { return self }
+    func variadic(y: Int, x: Int...)
+  }
+
+  struct B {
+    func someFunc(a: Int, c: String = "", d: String = "") {}
+  }
+
+  func overloaded() -> A { return A() }
+  func overloaded() -> B { return B() }
+
+  let localInt = 10
+  let localString = "Hello"
+
+  overloaded().someFunc(a: 2, #^ARG_NO_LABEL^#)
+
+  // ARG_NO_LABEL: Begin completions, 3 items
+  // ARG_NO_LABEL-DAG: Pattern/Local/Flair[ArgLabels]:     {#b: Int#}[#Int#]; name=b: Int
+  // ARG_NO_LABEL-DAG: Pattern/Local/Flair[ArgLabels]:     {#c: String#}[#String#]; name=c: String
+  // ARG_NO_LABEL-DAG: Pattern/Local/Flair[ArgLabels]:     {#d: String#}[#String#]; name=d: String
+  // ARG_NO_LABEL: End completions
+
+  overloaded().someFunc(a: 2, b: #^ARG_LABEL^#)
+
+  // ARG_LABEL: Begin completions
+  // ARG_LABEL-DAG: Decl[LocalVar]/Local:                          localString[#String#]; name=localString
+  // ARG_LABEL-DAG: Decl[LocalVar]/Local/TypeRelation[Identical]:  localInt[#Int#]; name=localInt
+  // ARG_LABEL: End completions
+
+  overloaded().someFunc(a: 2, c: "Foo", #^ARG_NO_LABEL2^#)
+
+  // ARG_NO_LABEL2: Begin completions, 1 item
+  // ARG_NO_LABEL2: Pattern/Local/Flair[ArgLabels]:     {#d: String#}[#String#]; name=d: String
+  // ARG_NO_LABEL2: End completions
+
+  overloaded().someFunc(a: 2, wrongLabel: "Foo", #^ARG_NO_LABEL_PREV_WRONG^#)
+
+  // ARG_NO_LABEL_PREV_WRONG: Begin completions, 2 items
+  // ARG_NO_LABEL_PREV_WRONG-DAG: Pattern/Local/Flair[ArgLabels]:     {#c: String#}[#String#]; name=c: String
+  // ARG_NO_LABEL_PREV_WRONG-DAG: Pattern/Local/Flair[ArgLabels]:     {#d: String#}[#String#]; name=d: String
+  // ARG_NO_LABEL_PREV_WRONG: End completions
+
+  overloaded().someFunc(a: 2, d: "Foo", #^ARG_NO_LABEL_OUT_OF_ORDER^#)
+  // ARG_NO_LABEL_OUT_OF_ORDER: Begin completions
+  // ARG_NO_LABEL_OUT_OF_ORDER-NOT: name=d: String
+  // ARG_NO_LABEL_OUT_OF_ORDER: Pattern/Local/Flair[ArgLabels]:     {#c: String#}[#String#]; name=c: String
```

Comment:
This is because there's an existing fix in the matching logic to reorder out-of-order arguments, and it binds the completion to `c:`. I felt ok with it because:
- it's consistent with regular compilation. Both use the same logic so if you accept the completion you get an out-of-order diagnostic with a fixit to move the `c:` arg in front of `d:`. You'll only get the completion suggestion when you're going to get that fixit.
- if you're trying to complete here you probably either think the arguments are in a different order to what they are, or don't know what arguments/order is expected and are exploring. I think showing something in that case may be useful. E.g with the Package() call in swift package manifest files the args (`targets:`, `products:`, `dependencies:`, `swiftLanguageVersions:`, etc.) are all defaulted and must be ordered correctly because the language requires it, but there's no particularly intuitive way to know where among existing arguments you should try completing when trying to add in one of the previously defaulted ones in an existing call.

I feel like the fix itself makes sense for completion generally (e.g. I think `someFunc(c: "blah", a: 2, #^HERE^#)` should still suggest `d:` and `someFunc(b: 2, a: 2). #^HERE^#` should still suggest `someFunc()` and `variadic()` even though neither compiles) but feel less strongly when an unlabelled completion argument is itself the out-of-order argument, as in this test. Do you think we should special-case disallow the fix in that case?

Reply:
Given your explanation, I think the behavior implemented in the PR is good, particularly the `Package.swift` case convinced me because I can never remember the correct argument order. Let’s keep it like this because it’s good for exploration.